backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/TaintVulnerabilityTrackingService.java

```diff
@@ -246,7 +246,7 @@ private static TaintVulnerabilityDto toDto(ServerTaintIssue serverTaintIssue, Pr
       serverTaintIssue.getMessage(),
       translation.serverToIdePath(serverTaintIssue.getFilePath()),
       serverTaintIssue.getCreationDate(),
-      isMQRMode && cleanCodeAttribute != null ? Either.forRight(new MQRModeDetails(cleanCodeAttribute, impacts))
+      isMQRMode && !impacts.isEmpty() ? Either.forRight(new MQRModeDetails(cleanCodeAttribute, impacts))
```

Comment:
When connected to SonarCloud and fetching Taint rule details, we get the Clean Code attribute but not the impacts :D So checking for impacts is a more robust way of telling which mode to use.

Reply:
Why are you checking for both `isMQRMode` && `impacts.isEmpty`? `isMQRMode` should be sufficient if you get it from the service

Reply:
Unfortunately, `isMQRMode` check does not seem sufficient. With SonarCloud, taint vulnerabilities don't seem to have `impacts` set at all.

I think there is something with protobuf mapping, did not have time to check but anyway this additional check guarantees that we get the correct data.

Reply:
Maybe add a TODO on this line to not forget about investigating this, or just a comment. In the meantime I guess we can go with this

Reply:
I'd also keep the `cleanCodeAttribute != null`, to avoid the warning